### ISSUE TITLE
PROD - Unable to replace a layer

### ISSUE THREAD

**Issue by meyerch** [geosolutions-it/unesco-ihp#145]
When trying to replace a layer by another one, we end up with an error message. 
[We tried to replace http://ihp-wins.unesco.org/layers/geonode:popwwcoll , both using a regular contributor account and an admin account. We had the problem each time]  

At the end of the upload of the replacement layer, we have the following message:
![replacelayer2.png](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59c90ce9c2355f0011ed2e90/125516c66e82c728ace21e0d46fa918a2f8380ac83e8a646b60fb2d86b56357c2301a03f32dfa17de6437f555a5b18fa105d5d41aaf98928a6b23c718f5a09ffcd7f1d2598694ef1e1c84afd13b0328a488fb7b9894cc5ea3e27ef888846360c48.png)

**Comment by giohappy** [comment]
well received @meyerch we're already working on that.

**Comment by afabiani** [comment]
@meyerch this should be fixed now.

**Comment by meyerch** [comment]
trying to replace the same layer, I got this error message:
![replace1.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59c90ce9c2355f0011ed2e90/125516c66e82c728ace21e0d46fa918a2f8380acdeabf301b44fff9e2c4533317346f6727181b77deb447d005a5d4bfa41585c44f8f9882ea0b42e6c981e5ab38532022d8c6853beb2901eb60bbb25884f82b1ba894dc3ee3c23f5c6.PNG)

**Comment by afabiani** [comment]
Hi @meyerch strange error, I'll try again with several inputs. At a first glance it looks something related to the DB (maybe the encoding or something). Is this happening with all the layers? Any chance to have the files you used for this test?

**Comment by meyerch** [comment]
Yes I am going to send you the file, and try with other layers in the meantime

**Comment by meyerch** [comment]
I tried to replace this one as well: http://ihp-wins.unesco.org/layers/geonode:wash_deaths_dalys2_1/replace 

and ended up with the following message as well :

![replace2.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59c90ce9c2355f0011ed2e90/125516c66e82c728ace21e0d46fa918a2f8380acddabf301b44fff9e2c4533317346f6727181b77deb447d005a5d4bfa41585c44f8f9882ea0b42e6c981e5ab38532022d8c6853beb2901eb60bbb25884f82b1bb8b49c3eb3f22f5c6.PNG)

Interestingly, now that I try to reopen the pages of both layers, I end up with a viewer that is max-zoomed in the middle of the Indian ocean (close to South Africa). The location of the zoom is the same for both layers

![replace3.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59c90ce9c2355f0011ed2e90/125516c66e82c728ace21e0d46fa918a2f8380acdcabf301b44fff9e2c4533317346f6727181b77deb447d005a5d4bfa41585c44f8f9882ea0b42e6c981e5ab38532022d8c6853beb2901eb60bbb25884f82b1bb8845c4e83f24f5c6.PNG)

**Comment by meyerch** [comment]
I tried with 2 layers very recently uploaded (http://ihp-wins.unesco.org/layers/geonode:greencons and http://ihp-wins.unesco.org/layers/geonode:bluecons) and it failed as well. I got the same error message.

**Comment by afabiani** [comment]
@meyerch about the last two layers, I see them working. Did you replaced them?

**Comment by meyerch** [comment]
@afabiani  I tried but it failed. I have not touched them since.

**Comment by giohappy** [comment]
dear @meyerch we tried substituting the two layers with one from us and it worked correctly. Can you give us the layer you're trying to set in place of the current?

**Comment by meyerch** [comment]
Dear @giohappy , 

I am out of office today and do not have access to the 2 layers i tried to replace for now. 

I was able to do the test again with other layers (http://ihp-wins.unesco.org/layers/geonode:categorytwocenters & http://ihp-wins.unesco.org/layers/geonode:c2c_chairs2, using the account of each layer's responsble person), and I no longer have an error message. 

However, I came accross another issue: apparently, the system does not actually replace the old data with the new ones, but somehow merge the 2 together. 
For instance, to do the replacement test, I first edited the new layer in QGIS, with a much more detailed attribute table.
![replacement1.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59c90ce9c2355f0011ed2e90/125516c66e82c728ace21e0d46fa918a2f8380ac82ecb157f51fbcd6395428792c07b7692ac7b671be462f545b0918fd445a5a12fcf98a2ba2b63d758c5e19f1c32f0e2a92624cf5f2d75dad0ba8218c4584b3bf8b4ec0e03420e68b864b2c53.PNG)

The original layer on WINS had the following fields in the attribute table: Centre; City; Country; Region; Long; Lat; Contact; Details. 
THe new layer contains the following information: Type; Name; University; Address; City; Country; Region; Long; Lat; Contact; Mail; Phone; Tests

Once replaced, only the Name of the fields from the old layer are kept, and filled with information from matching Field (= same name) from the new layer. The additional new fields are just discarded, while some old fields (= Center; Details) are blank because the new layer does not have fields with the same name. 

![replacement2.PNG](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/59c90ce9c2355f0011ed2e90/125516c66e82c728ace21e0d46fa918a2f8380ac82ecb157f61fbcd6395428792c07b7692ac7b671be462f545b0918fd445a5a12fcf98a2ba2b63d758c5e19f1c32f0e2a92624cf5f2d75dad0ba821834584b3bf8b4ec0e13d21e08e874a2c53.PNG)

So basically: there is still an issue with the replacement of layer. 

I hope this is understandable.
